src/ui/public/saved_object_table/item_table_actions.js

```diff
@@ -0,0 +1,94 @@
+import { ItemSelector } from './item_selector';
+import { ItemSorter } from './item_sorter';
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+import { TITLE_COLUMN_ID } from './get_title_column';
+
+export class ItemTableActions {
+  static toggleItem(itemTableState, item) {
+    if (itemTableState.isItemSelected(item)) {
+      ItemSelector.deselectItem(itemTableState, item);
+    } else {
+      ItemSelector.selectItem(itemTableState, item);
+    }
+  }
+
+  static toggleAll(itemTableState) {
+    if (itemTableState.areAllItemsSelected()) {
+      ItemSelector.deselectAll(itemTableState);
+    } else {
+      ItemSelector.selectAll(itemTableState);
+    }
+  }
+
+  static replaceItems(itemTableState, newItems) {
+    itemTableState.items = newItems;
+
+
+    ItemSelector.deselectAll(itemTableState);
+    itemTableState.items = ItemSorter.sortItems(
+      itemTableState.items,
+      itemTableState.currentSortedColumn || TITLE_COLUMN_ID,
+      itemTableState.currentSortOrder === SortOrder.NONE ? SortOrder.ASC : itemTableState.currentSortOrder);
+    this.calculateItemsOnPage(itemTableState);
+  }
+
+  static calculateItemsOnPage(itemTableState) {
+    itemTableState.pager.setTotalItems(itemTableState.items.length);
+    const startItemIndex = itemTableState.pager.startItem - 1;
+    const endItemIndex = itemTableState.pager.endItem;
+    itemTableState.pageOfItems = itemTableState.items.slice(startItemIndex, endItemIndex);
+  }
+
+  static sort(itemTableState, property) {
+    if (itemTableState.currentSortedColumn === property) {
+      itemTableState.currentSortOrder = ItemSorter.getFlippedSortOrder(itemTableState.currentSortOrder);
+    } else {
+      itemTableState.currentSortedColumn = property;
+      itemTableState.currentSortOrder = SortOrder.ASC;
+    }
+
+    itemTableState.items = ItemSorter.sortItems(
+      itemTableState.items,
+      itemTableState.currentSortedColumn,
+      itemTableState.currentSortOrder);
+    ItemSelector.deselectAll(itemTableState);
+    this.calculateItemsOnPage(itemTableState);
+  }
+
+  static turnToNextPage(itemTableState) {
+    ItemSelector.deselectAll(itemTableState);
+    itemTableState.pager.nextPage();
+    this.calculateItemsOnPage(itemTableState);
+  }
+
+  static turnToPreviousPage(itemTableState) {
+    ItemSelector.deselectAll(itemTableState);
+    itemTableState.pager.previousPage();
+    this.calculateItemsOnPage(itemTableState);
+  }
+
+  static filter(itemTableState, filter, service) {
```

Comment:
(Yellow flag) This function has a high degree of complexity. As someone trying to understand it, I will have to look at the implementation of itemTableState and service to understand how they work together. I'll also have to search the codebase to understand why `filter` is assigned to itemTableState.

I think we can look at the other functions here and assess them for complexity in a similar way.